_shared/project/conf/supervisord-dev.conf

```diff
@@ -1,29 +1,32 @@
 [supervisord]
 nodaemon = true
 silent = true
-{% if cookiecutter.get("_directory") == "pyapp" %}
 
-[program:app]
-command=python3 -m {{ cookiecutter.package_name }}
-stdout_events_enabled=true
-stderr_events_enabled=true
-stopsignal = KILL
-stopasgroup = true
+{% if include_exists("supervisor.json") %}
+{% set programs = include_json("supervisor.json").get("programs", {}) %}
+{% else %}
+{% if cookiecutter._directory == 'pyramid-app' %}
+{% set programs = {"web": {"dev": true, "prod": true, "settings": {"command": "gunicorn --bind :{{ cookiecutter.port }} --workers 1 --reload --timeout 0 --paste conf/development.ini"}}} %}
+{% else %}
+{% set programs = {"app": {"dev": true, "prod": true, "settings": {"command": "python3 -m h_periodic"}}} %}
 {% endif %}
-{% if cookiecutter.get("_directory") == "pyramid-app" %}
-
-[program:web]
-command=gunicorn --bind :{{ cookiecutter.port }} --workers 1 --reload --timeout 0 --paste conf/development.ini
-stdout_events_enabled=true
-stderr_events_enabled=true
-stopsignal = KILL
-stopasgroup = true
 {% endif %}
-{% if include_exists("conf/supervisord-dev.conf") %}
 
-{{ include("conf/supervisord-dev.conf") -}}
-{% endif %}
+{%- for program_name in programs %}
+{% if programs[program_name].get("dev") %}
+[program:{{ program_name }}]
+{% set program_settings = programs[program_name].get("settings", {}) %}
+command={{ program_settings.pop("command")}}
+{% set default_program_settings = {"stdout_events_enabled": "true", "stderr_events_enabled": "true", "stopsignal": "KILL", "stopasgroup": "true"} %}
+{% for setting_name in default_program_settings.keys()|sort %}
+{{ setting_name }}={{ program_settings.pop(setting_name, default_program_settings[setting_name]) }}
+{% endfor %}
+{% for setting_name, setting_value in program_settings.items() %}
+{{ setting_name }}={{ setting_value }}
+{% endfor %}
 
+{% endif %}
+{% endfor %}
```

Comment:
Ugh, maybe there's a nicer way to write this template code? But Jinja2 doesn't make it easy